examples/seismic/model.py

```diff
@@ -377,14 +405,22 @@ def __init__(self, origin, spacing, shape, space_order, vp, nbpml=20,
         else:
             self.theta = 0
 
-        if phi is not None:
+        if phi is not None and self.grid.dim>2:
             if isinstance(phi, np.ndarray):
                 self.phi = Function(name="phi", grid=self.grid, space_order=space_order)
                 initialize_function(self.phi, phi, self.nbpml)
             else:
                 self.phi = phi
         else:
-            self.phi = 0
+            self.phi = None
+
+    def physical_params(self, **kwargs):
+        """
+        Return all set physical parameters
+        """
+        known = [getattr(self, i) for i in self._physical_parameters]
+        return {i.name: kwargs.get(i.name, i) or i for i in known
```

Comment:
is ``or i`` really necessary here?

Reply:
Yes it is. If `i` is in `kwargs` as `i=None` you would end up with `i.name:None` that you don't want.

Reply:
but can ``i`` be ``None`` if you're testing for ``isinstance(i, Function)`` ?

Reply:
reminder

Reply:
`kwargs` will always contain all the physical parameter but some may be `None`, so the `or i` is required to get back the default physical parameter.